### DISCUSSION TITLE
Use of GEO1 and GEO2

### DISCUSSION THREAD

**Discussion by richard-sisson** [australiangeomechanics/agsformat#52] [Q&A] [answered]
Hi,

Can you please explain the use of GEO1 and GEO2 in version 1.2.

It appears that these are replacements for GEOL_GEOL and GEOL_GEO2 used in other AGS formats (i.e. UK).

Thanks
Richard

**Comment by SolidRockProgrammer** [comment, marked as answer]
Hi Richard
Our group description is lacking.
Your inference is correct.  We have split out geology/geotechnical units to their own groups, as it is practice in Australia to separate factual/field data on GEOL and interpretative data like units.
regards
Phil

**Reply by richard-sisson** [reply]
Hi Phil,
That's fair enough.
Are depths (i.e. TOP and/or BASE) expected to match a record in the GEOL group, or can the change occur within a GEOL layer?
Thanks
Richard

**Comment by SolidRockProgrammer** [comment]
They are independent of GEOL. We expect people to transmit consolidated rows, i.e. don't have two rows of same unit boarding on each other.

**Reply by richard-sisson** [reply]
Understood, but is it expected the depths in GEO1/GEO2 match a layer boundary in GEOL?

**Reply by SolidRockProgrammer** [reply]
No, but I expect often they would

**Comment by RobDayArup** [comment]
I have noticed that AGS3.1 files that have been generated from different versions of the RTA gINT AGS3.1 files by multiple companies over multiple years have a problem with data loss and duplication of Geology (GEO1 and GEO2) data.  The reason for this appears to be as follows.

From the beginning, RTA 3.1 gINT files in 2010 already had GEOL_GEOL and GEOL_GEO2 fields from the GEOL group split out into separate groups within gINT called GEOLOGY_PROJECT_UNIT and GEOLOGY_PROJECT_UNIT_2 and deleted these fields from the GEOL group, (i.e. as adopted in AGS4.1AU1.2).  This allowed these to have different depth ranges to the depths in GEOL and a project library table was used to populate these fields as type PA.  The correspondence files merged these back into the GEOL group for AGS3.1 export using SQL to match up depths in the GEOL table, and split these back out into GEOLOGY_PROJECT_UNIT and GEOLOGY_PROJECT_UNIT_2 when re-importing.  Hence there were only two fields for storing Geology information as is the case for AGS3.1, AGS4.1 and AGS4.1AU1.2.

The problem is over the years the RTA gINT Data Template was also modified for better compatibility with UK AGS format for non-RTA projects to also have GEOL_GEOL and GEOL_GEO2 in the GEOL group.  All four had their own Library tables so you could effectively have FOUR project geologies.  It appears that in many instances various companies have internally used GEOLOGY_PROJECT_UNIT and GEOLOGY_PROJECT_UNIT_2 as intended, but sometimes also use GEOL_GEOL and GEOL_GEO2 as well on the same project.

The key issue arising from this is that during AGS3.1 export there are many cases where only the GEOL_GEOL and GEOL_GEO2 from GEOL have been exported and data from GEOLOGY_PROJECT_UNIT and GEOLOGY_PROJECT_UNIT_2 has been lost.

The fix for the lost data in AGS4.1AU1.2 would be to export the groups GEOLOGY_PROJECT_UNIT and GEOLOGY_PROJECT_UNIT_2 to the AGS4.1AU1.2 groups GEO1 and GEO2.

BUT AGS4.1AU1.2 Deprecates GEOL_GEOL and GEOL_GEO2 fields from the GEOL table so there is no place to import any data that was stored in these fields in the gINT RTA Data Template into AGS4.1AU1.2.  To avoid data loss when importing historical projects, GEOL_GEOL and GEOL_GEO2 need to NOT be Deprecated in the AGS files for projects containing this data.

**Comment by SolidRockProgrammer** [comment]
Hi Rob

We won't have two headings for the same thing in AGS 4.1.1 AU.  

Much of what you discussed is a internal data management or software problem, and it is your choice how to deal with it.  You can write a gINT gci file to import GEOL.GEOL_GEOL from and AGS 3.1 RTA 1.1 ags file or Datgel gINT AGS RTA database (it is Datgel's IP, it didn't come with gINT) into a table for GEO1 in a gINT database. I can't talk to how to do this in other software. You can have what ever fields you want in your database.

regards

Phil

**Comment by RobDayArup** [comment]
Phil
I agree that this is not an issue for AGS 4.1.1 AU and that deprecation on GEOL_GEO1 and GEOL_GEO2 is correct.  The observation I have made above is something I have observed from testing multiple AGS files from different companies received on different projects that I have been looking at to write our import scripts.  It is up to individual organisations how they deal with this, but I was just alerting people to the problem so they can work out strategies to deal with it.  For new projects to AGS 4.1.1 AU I absolutely agree there are only two geology fields.  In our case we have left the deprecated fields available in the software database for old/legacy projects.
